server/gamehand.cpp

```diff
@@ -197,12 +197,8 @@ static struct tile *place_starting_unit(struct tile *starttile,
   /* Expose visible area. */
   map_show_circle(pplayer, ptile, game.server.init_vis_radius_sq);
 
-  if (utype != NULL) {
-    (void) create_unit(pplayer, ptile, utype, FALSE, 0, 0);
-    return ptile;
-  }
-
-  return NULL;
+  (void) create_unit(pplayer, ptile, utype, FALSE, 0, 0);
```

Comment:
False positive, `utype` can be `NULL` if `ptype` is `NULL`, because `crole_to_unit_type` can return `NULL`.

Reply:
Coverity doesnt have false positives in such cases
struct tile *ptile = NULL;
 utype is set to something 

  if (utype != NULL) {
       ** some iteartion combo and ptile is set here
  }

  if (ptile == NULL) {
    ** if utype omited ealier iteration combo(beacause was NULL) ptile would be NULL also and would return here
    return NULL;
  }